crypto/src/asn1/DerObjectIdentifier.cs

```diff
@@ -206,18 +207,16 @@ private byte[] GetContents()
         internal static DerObjectIdentifier CreatePrimitive(byte[] contents, bool clone)
         {
             int hashCode = Arrays.GetHashCode(contents);
-            int first = hashCode & 1023;
 
-            lock (cache)
-            {
-                DerObjectIdentifier entry = cache[first];
-                if (entry != null && Arrays.AreEqual(contents, entry.GetContents()))
-                {
-                    return entry;
-                }
+            DerObjectIdentifier entry = cache.GetOrAdd(hashCode, new DerObjectIdentifier(contents, clone));
```

Comment:
No point caching it if you parse it each time.
Add `_ =>` in `GetOrAdd`.

```csharp
DerObjectIdentifier entry = cache.GetOrAdd(hashCode, _ => new DerObjectIdentifier(contents, clone));
```

Reply:
Got lost when transferring from the test branch, fixed it.